server/src/main/java/io/crate/metadata/pgcatalog/PgCatalogTableDefinitions.java

```diff
@@ -67,12 +67,15 @@ public PgCatalogTableDefinitions(InformationSchemaIterables informationSchemaIte
             informationSchemaIterables::pgClasses,
             (user, t) -> user.hasAnyPrivilege(Privilege.Clazz.TABLE, t.ident.fqn())
                          // we also need to check for views which have privileges set
-                         || user.hasAnyPrivilege(Privilege.Clazz.VIEW, t.ident.fqn()),
+                         || user.hasAnyPrivilege(Privilege.Clazz.VIEW, t.ident.fqn())
+                         || isUnderSystemSchemas(t.ident.fqn()),
             pgCatalogSchemaInfo.pgClassTable().expressions()
         ));
         tableDefinitions.put(PgProcTable.IDENT, new StaticTableDefinition<>(
             informationSchemaIterables::pgProc,
-            (user, f) -> user.hasAnyPrivilege(Privilege.Clazz.SCHEMA, f.functionName.schema()),
+            (user, f) -> user.hasAnyPrivilege(Privilege.Clazz.SCHEMA, f.functionName.schema())
+                         || isUnderSystemSchemas(f.functionName.displayName())
+                         || isBuiltinFunction(f.functionName.displayName()),
```

Comment:
Instead of doing the `isBuiltinFunction` test on the display name this could check if there's a schema on the `functionName`. Or we could introduce a `isBuiltIn()` property on `FunctionName`.

This also doesn't account for built-in functions within the `pg_catalog` schema. So I think there should be a `isBuiltIn` where there is either no schema or the schema matches one of the built-in system schemas.

Reply:
I have created isBuiltin() on `FunctionName` which checks no schema and the system schemas.

Also, I thought I could apply the similar logic to `isUnderSystemSchemas(..)` and moved that into `RelationName`. Please have a look.